File: MVVM初次使用/MVVM初次使用/ViewController.m
//
//  ViewController.m
//  MVVM初次使用
//
//  Created by Tianjian on 16/3/30.
//  Copyright © 2016年 oneyd.me. All rights reserved.
//

#import "ViewController.h"
#import "TableViewDataSource.h"
#import "TableViewCell.h"

@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>

@property (strong,nonatomic) UITableView *tableView;
@property (strong,nonatomic) NSArray<NSDictionary *> *arrCtrlNames;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _arrCtrlNames = @[@{@"title":@"dataSource&&delegate分离",@"class":@"Demo01VC"}];
    
    self.tableView = ({
        UITableView *ettblView = [[UITableView alloc]initWithFrame:CGRectMake(0, 0,CGRectGetWidth(self.view.frame), CGRectGetHeight(self.view.frame)-20) style:UITableViewStylePlain];
        ettblView.delegate = self;
        ettblView.dataSource = self;
        [ettblView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cellIndentifier"];
        [self.view addSubview:ettblView];
        ettblView;
    });
    
    //UIViewController *classVC = [NSClassFromString([_arrCtrlNames lastObject][@"class"]) new];
    //[self.navigationController pushViewController:classVC animated:YES];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return _arrCtrlNames.count;
}
- (UITableViewCell*)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cellIndentifier"];
    cell.textLabel.text   = [NSString stringWithFormat:@"%@",_arrCtrlNames[indexPath.row][@"title"]];
    return cell;
}
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    NSDictionary *dict = _arrCtrlNames[indexPath.row];
    UIViewController *classVC = [NSClassFromString(dict[@"class"]) new];
    [self.navigationController pushViewController:classVC animated:YES];
}

@end

File: SwiftTableView/SwiftTableView/ViewController.swift
//
//  ViewController.swift
//  SwiftTableView
//
//  Created by Tianjian on 16/1/17.
//  Copyright © 2016年 oneyd.me. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {

    var tableView :UITableView?
    var items = ["武汉","上海","北京","深圳","广州","重庆","香港","台海","天津"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        initTableView()
    }
    
    func initTableView(){
        self.tableView = UITableView(frame:self.view.frame,style: UITableViewStyle.Plain)
        self.tableView!.delegate = self;
        self.tableView!.dataSource = self;
        self.tableView!.registerClass(UITableViewCell.self, forCellReuseIdentifier: "cell")
        self.view.addSubview(self.tableView!)
    }

    //1.numberOfLines
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.items.count
    }

    //2.
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView .dequeueReusableCellWithIdentifier("cell", forIndexPath: indexPath) as UITableViewCell
        
        var row = indexPath.row as Int
        
        cell.textLabel?.text = self.items[row]
        
        return cell;
    }
}

